### ISSUE TITLE
[Twilio SMS] Proactive SMS messages to users that have not previously interacted with the bot

### ISSUE THREAD

**Issue by jreiter** [Microsoft/BotBuilder#3653]
## Bot Info 
* SDK Platform: Node.js
* SDK Version: 3.11.0
* Active Channels: SMS (Twilio)
* Deployment Environment: Other (AWS)

## Issue Description
I'd like to be able to send proactive messages, from my bot, to users who have agreed to receive SMS messages.  All of the examples I've found, including [this doc](https://docs.microsoft.com/en-us/bot-framework/nodejs/bot-builder-nodejs-proactive-messages), assume that the user has previously interacted with the bot, and that you can use a saved address object from the session.  In the case of SMS, I have the user's ID (their phone number), and I want to begin a dialog based on that information.  Is this possible?  I've seen some examples in other Github issues, but all of them seem to start with the prerequisite of the user having initiated conversation with the bot at some point.

## Code Example
Based on other examples, I've been trying things like the below (I'm registering a dialog at `'/'` elsewhere).

```javascript
bot.beginDialog(
  {
    user: { id: '+1234567890' },
    bot: { id: '+19876543210' },
    channelId: 'sms',
    serviceUrl: 'https://sms.botframework.com',
    useAuth: false
  },
  '/'
);
```

## Expected Behavior
User receives an SMS message, beginning a dialog with the bot.

## Actual Results

```
ERROR: ChatConnector: startConversation - error starting conversation.
Error: Failed to start conversation: no conversation ID returned.
    at /Users/reiterj/workspace/zipscript/big/node_modules/botbuilder/lib/bots/ChatConnector.js:225:35
    at Request._callback (/Users/reiterj/workspace/zipscript/big/node_modules/botbuilder/lib/bots/ChatConnector.js:528:37)
    at Request.self.callback (/Users/reiterj/workspace/zipscript/big/node_modules/request/request.js:186:22)
    at emitTwo (events.js:125:13)
    at Request.emit (events.js:213:7)
    at Request.<anonymous> (/Users/reiterj/workspace/zipscript/big/node_modules/request/request.js:1163:10)
    at emitOne (events.js:115:13)
    at Request.emit (events.js:210:7)
    at IncomingMessage.<anonymous> (/Users/reiterj/workspace/zipscript/big/node_modules/request/request.js:1085:12)
    at Object.onceWrapper (events.js:314:30)
```

**Comment by jreiter** [comment]
I actually was able to get this working by adding a `conversation` property to my `address` param, setting the `id` to my user's phone number.  Are there any problems with this approach?  Is there a better way to do it?

```javascript
bot.beginDialog(
  {
    user: { id: '+1234567890' },
    bot: { id: '+19876543210' },
    conversation: { id: '+1234567890' },
    channelId: 'sms',
    serviceUrl: 'https://sms.botframework.com'
  },
  '/'
);
```

**Comment by nrobert** [comment]
I think that's the right way, I did it a few months ago with C# implementation.
One of the problem with this approach that you must keep in mind is the fact that in some countries you are not allowed to send marketing SMS if your recipient is not "opt in"

**Comment by jreiter** [comment]
@nrobert thanks.  Yes, we have an opt-in process that is just external to SMS.

I guess the thing that concerns me is setting the conversation ID myself, and the fact that this only works when I set the ID to a specific value (the outgoing phone number).  It makes me worry that 1.) there doesn't appear to be any documentation on this, and 2.) the conversation ID being the same as the outgoing number may be an implementation detail that I've stumbled upon, and it could therefore change in the future without notice.
